animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/Main.java

```diff
@@ -80,16 +82,21 @@ public static void main( String[] args )
                 continue;
             }
 
-            files.add(new File(args[i]));
+            try {
+                files.add(Paths.get(new URI(args[i]))); // attempt to treat it as an URI
```

Comment:
Might be good to verify that the scheme of the URI matches one of the expected ones. Otherwise a mistyped path (or ambiguity we have not considered yet) could lead to usage of file system the user did not intend to use.
It would then probably be best to refactor this code to a separate method:
```java
private static Path getPath(String s) {
    try {
        URI uri = new URI(s);
        String scheme = uri.getScheme();
        // Only allow certain schemes to prevent treating (mistyped) file
        // path unintentionally as URI
        if (scheme.equalsIgnoreCase("file") || scheme.equalsIgnoreCase("jrt")) {
            return Paths.get(uri);
        }
        // Fall through
    } 
    catch (URISyntaxException e) {
        // Fall through; probably not a URI but a file path
    }
    
    return Paths.get(s);
}
```

Reply:
I have integrated your suggestions.

Reply:
Thanks!